File: Campus_Services_Portal/Models/Entities/Complaint.cs
using Campus_Services_Portal.Models.Enums;

namespace Campus_Services_Portal.Models.Entities
{
    public class Complaint
    {
        public int Id { get; set; }

        public int StudentId { get; set; }

        public int ComplaintCategoryId { get; set; }

        public string Description { get; set; } = string.Empty;

        public ComplaintStatus Status { get; set; }
            = ComplaintStatus.Pending;

        public string? ResolutionNote { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Relationships
        public Student Student { get; set; } = null!;

        public ComplaintCategory ComplaintCategory { get; set; } = null!;
    }
}

File: campusx-frontend/src/app/core/models/complaint.ts
// =====================================================
// COMPLAINT STATUS
// =====================================================

export type ComplaintStatus =
  | 'Pending'
  | 'InProgress'
  | 'Resolved';


// =====================================================
// COMPLAINT CATEGORY
// =====================================================

export interface ComplaintCategory {

  id: number;

  name: string;

  description?: string;

}


// =====================================================
// COMPLAINT
// =====================================================

export interface Complaint {

  id: number;

  studentId: number;

  categoryId: number;

  categoryName?: string;

  description: string;

  status: ComplaintStatus;

  resolutionNote?: string;

  createdAt: string;

  updatedAt?: string;

}


// =====================================================
// CREATE COMPLAINT
// =====================================================

export interface CreateComplaint {

  categoryId: number;

  description: string;

}


// =====================================================
// CREATE COMPLAINT CATEGORY
// =====================================================

export interface CreateComplaintCategory {

  name: string;

  description?: string;

}


// =====================================================
// UPDATE COMPLAINT STATUS
// =====================================================

export interface UpdateComplaintStatus {

  status: ComplaintStatus;

  resolutionNote?: string;

}